pkg/library/libraryinputresources/prune_mustgather.go

```diff
@@ -41,6 +44,33 @@ func WriteRequiredInputResourcesFromMustGather(ctx context.Context, inputResourc
 	return errors.Join(errs...)
 }
 
+func listUnnecessaryFilesInMustGather(mustGatherDir string, inputResources []*Resource) ([]string, error) {
```

Comment:
the report shouldn't be filenames (recall that we can have files that multiple resources in them), the report should be group, resource, namespace, name values.  with potentially a filename as a courtesy.